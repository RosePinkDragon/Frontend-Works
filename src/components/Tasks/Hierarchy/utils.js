import * as d3 from "d3";

export const graph = (data, element) => {
  const svg = d3
    .select(element)
    .append("svg")
    .attr("width", 600)
    .attr("height", 600)
    .append("g")
    .attr("transform", "translate(50,50)");

  const dataStructure = d3
    .stratify()
    .id((d) => d.id)
    .parentId((d) => d.reporting_id)(data);

  const information = d3.tree().size([500, 300])(dataStructure);

  const circles = svg
    .append("g")
    .selectAll("circle")
    .data(information.descendants());

  circles
    .enter()
    .append("circle")
    .attr("cx", (d) => d.x)
    .attr("cy", (d) => d.y)
    .attr("r", 5);

  const connections = svg
    .append("g")
    .selectAll("path")
    .data(information.links());

  connections
    .enter()
    .append("path")
    .attr("d", function (d) {
      return (
        "M" +
        d.source.x +
        "," +
        d.source.y +
        " C " +
        d.source.x +
        "," +
        (d.source.y + d.target.y) / 2 +
        " " +
        d.target.x +
        "," +
        (d.source.y + d.source.y) / 2 +
        " " +
        d.target.x +
        "," +
        d.target.y
      );
    });

  const names = svg
    .append("g")
    .selectAll("text")
    .data(information.descendants());
  names
    .enter()
    .append("text")
    .text((d) => d.data.id.split("/")[2])
    .attr("x", (d) => d.x + 7)
    .attr("y", (d) => d.y + 5);
};
