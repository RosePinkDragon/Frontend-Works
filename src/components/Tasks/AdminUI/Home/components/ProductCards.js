import React from "react";
import { productData } from "utils/HomeData";

const ProductCards = () => {
  console.log(productData.length);

  return (
    <div className="cardWrap">
      {productData.map(({ name, products }) => (
        <div className="card" key={name}>
          <h3>{name}</h3>
          <h5>{products.length}</h5>
        </div>
      ))}
    </div>
  );
};

export default ProductCards;
