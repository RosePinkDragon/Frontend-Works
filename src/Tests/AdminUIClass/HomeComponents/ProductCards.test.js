import ProductCards from "components/Tasks/AdminUIClass/Home/components/ProductCards";
import { productData } from "utils/HomeData";

import { mount } from "enzyme";

describe("ProductCards", () => {
  const wrapper = mount(<ProductCards />);
  it("renders product cards", () => {
    expect(wrapper.find(".card").length).toEqual(productData.length);
    expect(wrapper.find(".cardWrap").length).toEqual(1);
  });
  it("should render the Products Card Component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
