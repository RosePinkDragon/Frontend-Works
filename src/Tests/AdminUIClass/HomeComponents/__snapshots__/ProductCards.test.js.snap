// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCards should render the Products Card Component correctly 1`] = `
<ProductCards>
  <div
    className="cardWrap"
  >
    <div
      className="card"
      key="Passed Products"
    >
      <h3>
        Passed Products
      </h3>
      <h5>
        6
      </h5>
    </div>
    <div
      className="card"
      key="Upcoming Products"
    >
      <h3>
        Upcoming Products
      </h3>
      <h5>
        6
      </h5>
    </div>
    <div
      className="card"
      key="Total Products"
    >
      <h3>
        Total Products
      </h3>
      <h5>
        6
      </h5>
    </div>
  </div>
</ProductCards>
`;
